Fix #6168 by using `getBuiltin` in `levelType`. Proper error over crash.

andreasabel opened the pull request:
Fix #6168 by using `getBuiltin` in `levelType`. Proper error over crash.

andreasabel committed:
Fix #6168 by using `getBuiltin` in `levelType`. Proper error over crash.

jespercockx commented:
As I've said elsewhere I'm not sure if the introduction of `--no-load-primitives` was a good idea, so I'd hold off the merging of this PR until that issue is decided.

andreasabel commented:
This PR fixes just one case of undefined level builtin, it should be expanded to treat also the other level builtins...

andreasabel committed:
Re #6168: simplify `sortKit` (fallible) by using `getBuiltin`

andreasabel committed:
Fix #6168a: require all LEVEL builtins in levelType

This way, we make sure that we won't hit an internal error later when
some of the level builtins are missing.

In CheckInternal and Conversion we may use the unsafe `levelType'` as
here we presumably have seen levels before (e.g. in the type checker).

andreasabel commented:
I think I fixed this by requiring _all_ level BUILTINs in `levelType`.  So if you want to use LEVEL, you have to define also the other builtins.  
It was already the case that when you want to use TYPE, you also have to define the other sorts (PROP, STRICTSET, ....).
